File: projects/mcp-8004-enterprise/agents_implementation/onboarding-orchestrator-server.js
/**
 * onboarding-orchestrator-server.js
 *
 * MCP server for the BankOnboardingOrchestrator card.
 * The initiate_onboarding tool is a planning tool only — the actual on-chain
 * transactions are submitted by the orchestrator bridge, which watches for
 * the MCP response and translates it into contract calls.
 */

import { startMcpServer } from './mcp-server-base.js';

const tools = {

  initiate_onboarding: {
    description: 'Plan an institutional client onboarding flow initiation.',
    inputSchema: {
      type: 'object',
      properties: {
        flow_id:              { type: 'string' },
        client_address:       { type: 'string' },
        bank_aml_agent_id:    { type: 'string' },
        bank_credit_agent_id: { type: 'string' },
        bank_legal_agent_id:  { type: 'string' },
        hf_doc_agent_id:      { type: 'string' },
        hf_credit_agent_id:   { type: 'string' },
        hf_legal_agent_id:    { type: 'string' },
        trace_id:             { type: 'string' },
      },
      required: ['flow_id', 'client_address', 'bank_aml_agent_id', 'bank_credit_agent_id',
                 'bank_legal_agent_id', 'hf_doc_agent_id', 'hf_credit_agent_id', 'hf_legal_agent_id'],
    },
    handler(params) {
      const { flow_id, trace_id, ...rest } = params;
      if (!flow_id) throw new Error('flow_id required');
      console.log(`[orchestrator-server] [${trace_id ?? 'n/a'}] initiate_onboarding flow=${flow_id}`);

      // Return the plan; the bridge submits it on-chain
      return {
        flow_id,
        status:  'ready_to_initiate',
        agents:  {
          bank_aml_agent_id:    rest.bank_aml_agent_id,
          bank_credit_agent_id: rest.bank_credit_agent_id,
          bank_legal_agent_id:  rest.bank_legal_agent_id,
          hf_doc_agent_id:      rest.hf_doc_agent_id,
          hf_credit_agent_id:   rest.hf_credit_agent_id,
          hf_legal_agent_id:    rest.hf_legal_agent_id,
        },
      };
    },
  },
};

startMcpServer({ tools, resources: [], prompts: {}, serverLabel: 'onboarding-orchestrator-server' });

File: examples/mcp-8004-enterprise/agents_implementation_py/servers/onboarding_orchestrator_server.py
"""
onboarding_orchestrator_server.py — MCP server for the bank onboarding orchestrator.
Port of onboarding-orchestrator-server.js.

The initiate_onboarding tool is the entry point for the LangGraph runner.
It does planning/logging and returns flow metadata; actual on-chain transactions
are dispatched by onboarding_orchestrator_bridge.py.

Usage:
    python -m servers.onboarding_orchestrator_server [port]   # default 8013
"""

from __future__ import annotations

import logging
import sys

from servers.server_base import create_server, suspended_when_revoked
from shared.bounds_monitor_client import report_tool_call

logger = logging.getLogger(__name__)

mcp = create_server("onboarding-orchestrator-server")


@mcp.tool()
@suspended_when_revoked("initiate_onboarding")
async def initiate_onboarding(
    flow_id: str,
    client_address: str,
    bank_aml_agent_id: str = "0",
    bank_credit_agent_id: str = "1",
    bank_legal_agent_id: str = "2",
    hf_doc_agent_id: str = "7",
    hf_credit_agent_id: str = "8",
    hf_legal_agent_id: str = "9",
    trace_id: str = "",
) -> dict:
    """
    Plan and log an onboarding flow initiation.
    Returns flow metadata; the bridge dispatches on-chain transactions.
    """
    logger.info(
        "[orchestrator-server] [%s] initiate_onboarding flow=%s client=%s",
        trace_id or "n/a", flow_id, client_address,
    )
    report_tool_call("initiate_onboarding", success=True)

    return {
        "flow_id": flow_id,
        "status": "planned",
        "aml_request_id": None,
        "credit_request_id": None,
        "legal_request_id": None,
        "agents": {
            "bank_aml":    bank_aml_agent_id,
            "bank_credit": bank_credit_agent_id,
            "bank_legal":  bank_legal_agent_id,
            "hf_doc":      hf_doc_agent_id,
            "hf_credit":   hf_credit_agent_id,
            "hf_legal":    hf_legal_agent_id,
        },
    }


if __name__ == "__main__":
    logging.basicConfig(level=logging.INFO)
    port = int(sys.argv[1]) if len(sys.argv) > 1 else 8013
    logger.info("[onboarding-orchestrator-server] Starting on port %d", port)
    mcp.run(transport="streamable-http", host="0.0.0.0", port=port)
